packages/design-system/src/components/keyboard/index.tsx

```diff
@@ -55,50 +63,41 @@ const CLICKABLE_INPUT_TYPES = [
   'reset',
 ];
 
-const globalRef = createRef();
+const globalRef = createRef<HTMLElement>();
 
 function setGlobalRef() {
   if (!globalRef.current) {
     globalRef.current = document.documentElement;
   }
 }
 
-/**
- * @callback KeyEffectCallback
- * @param {KeyboardEvent} event Event.
- */
-
 /**
  * See https://craig.is/killing/mice#keys for the supported key codes.
- *
- * @param {Node|{current: Node}} refOrNode Node or reference to one.
- * @param {string|Array|Object} keyNameOrSpec Single key name or key spec.
- * @param {string|undefined} type Event type, either 'keydown', 'keyup', or undefined to automatically determine it.
- * @param {KeyEffectCallback} callback Callback.
- * @param {Array|undefined} deps The effect's dependencies.
  */
 function useKeyEffectInternal(
-  refOrNode,
-  keyNameOrSpec,
-  type,
-  callback,
-  deps = undefined
+  refOrNode: RefOrNode,
+  keyNameOrSpec: KeyNameOrSpec,
+  type: string | undefined,
+  callback: KeyEffectCallback,
+  deps: DependencyList
 ) {
   const { keys } = useContext(Context);
   //eslint-disable-next-line react-hooks/exhaustive-deps -- Pass through provided deps.
   const batchingCallback = useBatchingCallback(callback, deps || []);
```

Comment:
Looks like `useBatchingCallback` doesn't work with `(event) => ...` currently. Getting the following error:
```
Argument of type 'KeyEffectCallback' is not assignable to parameter of type '(...args: unknown[]) => unknown'.
  Types of parameters 'event' and 'args' are incompatible.
    Type 'unknown' is not assignable to type 'KeyboardEvent'.
```

Probably needs some adjustments there.